### PR TITLE
Expose command args for legate driver help. 

### PR DESCRIPTION
**Pull request by bryevdv**
This PR has prefactory work to allow the legate driver help to include command-line flags for known, installed downstream projects e.g. cunumeric.

In order to allow `cunumeric.ARGS` to importable form plain-python apps like the driver, an environment variable "override" can be set to suppress all the API imports. This is clunky, but is only utilized in the driver's `print_help` function

### CODE REVIEW

File: cunumeric/args.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2021-2022 NVIDIA Corporation
```
Comment (manopapad) [inline cunumeric/args.py:1]: I think for new files we want the copyright to start on the current year.

**Review by manopapad — APPROVED** [review]

File: cunumeric/args.py (hunk 1 above)
Comment (bryevdv) [inline cunumeric/args.py:1]:
```suggestion
# Copyright 2023 NVIDIA Corporation
```

Comment (bryevdv) [reply cunumeric/args.py]: Pretty sure some other OSS projects I work on get this entirely wrong, old habits...

**Comment by bryevdv** [comment]
seeing if a merge fixes the two persistent failures

**Pull request merged**